config.go

```diff
@@ -264,6 +281,63 @@ func (f *FlakeIDGeneratorConfig) Validate() error {
 	return nil
 }
 
+type InvalidationConfig struct {
+	maxToleratedMissCount *int
+	// ReconciliationIntervalSeconds is the time in seconds for the reconciliation task interval.
+	// Configuring a value of zero seconds disables the reconciliation task.
+	reconciliationIntervalSeconds *int
+	err                           error
+}
+
+func (pc InvalidationConfig) Clone() InvalidationConfig {
+	return InvalidationConfig{
+		maxToleratedMissCount:         pc.maxToleratedMissCount,
+		reconciliationIntervalSeconds: pc.reconciliationIntervalSeconds,
+		err:                           pc.err,
+	}
+}
+
+func (pc *InvalidationConfig) Validate() error {
+	if pc.err != nil {
+		return fmt.Errorf("hazelcast.Invalidation: %w", pc.err)
+	}
+	return nil
+}
+
+func (pc *InvalidationConfig) SetMaxToleratedMissCount(count int) {
+	if err := check.NonNegativeInt32Config(count); err != nil {
+		pc.err = fmt.Errorf("MaxToleratedMissCount: %w", err)
+		return
+	}
+	pc.maxToleratedMissCount = &count
+}
+
+func (pc InvalidationConfig) MaxToleratedMissCount() int {
+	if pc.maxToleratedMissCount == nil {
+		return defaultMaxToleratedMissCount
+	}
+	return *pc.maxToleratedMissCount
+}
+
+func (pc *InvalidationConfig) SetReconciliationIntervalSeconds(seconds int) {
+	if err := check.NonNegativeInt32Config(seconds); err != nil {
+		pc.err = fmt.Errorf("ReconciliationIntervalSeconds: %w", err)
+		return
+	}
+	if seconds != 0 && seconds < defaultMinReconciliationIntervalSeconds {
+		pc.err = fmt.Errorf("ReconciliationIntervalSeconds: must be 0 or greater or equal to 30: %w", pubhzerrors.ErrInvalidConfiguration)
```

Comment:
I prefer a more detailed error here like in Java: 

```
Reconciliation interval can be at least %s seconds if it is not zero, but %d was configured."
                            + " Note: Configuring a value of zero seconds disables the reconciliation task.
```

Reply:
I think that belongs to the documentation. Also the user can look up their own configuration to see what's wrong.

Reply:
Right but Java throws this exception see:
 https://github.com/hazelcast/hazelcast/blob/14ef0dbefa4a6fa00a50a8873aeea02ec6c3f5e4/hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/RepairingTask.java#L114

I prefer verbose errors in general

Reply:
We return this error when the user attempts `StartNewClientWithConfig`. If there's an error in the config, the client doesn't start and returns an error, so the user immediately knows where the problem is and how to fix that. That's in contrast with the Java client, where this exception is thrown only if the user tries to get a map with the near cache, so it may not be apparent what the fix is.

The documentation is missing for some of the configs here, and I agree that "Configuring a value of zero seconds disables the reconciliation task." is an important information. I will add that in the documentation PR.